src/transform/transform.ts

```diff
@@ -523,13 +523,44 @@ function createResponseType(codeModel: CodeModel, group: OperationGroup, op: Ope
       // add this response schema to the global list of response
       const responseSchemas = <Array<Schema>>codeModel.language.go!.responseSchemas;
       responseSchemas.push(firstResp.schema);
+    } else if (isLROOperation(op)) {
+      // add this response schema with LRO fields to the global list of responses by 
+      // replacing the previously added response with the same name
+      const responseSchemas = <Array<Schema>>codeModel.language.go!.responseSchemas;
+      for (let i = 0; i < responseSchemas.length; i++) {
+        if (responseSchemas[i].language.go!.name === firstResp.schema.language.go!.name) {
+          responseSchemas.splice(i, 1, firstResp.schema);
+        }
+      }
     }
   }
   // create pageable type info
   if (isPageableOperation(op)) {
     if (codeModel.language.go!.pageableTypes === undefined) {
       codeModel.language.go!.pageableTypes = new Array<PagerInfo>();
     }
+    // Add a poller
+    if (isLROOperation(op)) {
```

Comment:
Can we move this to the LRO+paging PR so all the changes are together?

Reply:
I added this since the paging LRO is the only one that checks this functionality and in order to build we need to have the poller definition. I could add a skip for adding poller fields if it's also a paging LRO, then I could remove this section. Should I skip them then?

Reply:
Ah got it, no let's leave it.

Reply:
Actually, looking more at this, we already have a `isLROOperation()` check on line 583 which should create the poller after creating the pager, so this seems redundant.  It was working before.

Reply:
So the problem is that the pagers implementation has a return inside of it, so the subsequent LRO work can sometimes we reached and other times not depending on whether or not the return is reached in the pager code.

Reply:
I had also thought of moving LRO creation above pager creation, but LRO creation now has a return in it as well. So we would have the same problem for creating a pager in the LRO.

Reply:
Actually the returns should be breaks.

Reply:
Looks like the returns are just to avoid adding duplicate pagers/pollers.  I think they could be replaced with some boolean flags?

Reply:
Yeah breaks + a flag I think.

Reply:
Yes will do 👍

Reply:
Done